merkle/hashers/tree_hasher.go

```diff
@@ -39,9 +39,16 @@ type MapHasher interface {
 	// A height of 0 indicates an empty leaf. The maximum height is Size*8.
 	// TODO(gbelvin) fully define index.
 	HashEmpty(treeID int64, index []byte, height int) []byte
-	// HashLeaf computes the hash of a leaf that exists.
+	// HashLeaf computes the hash of a leaf that exists.  This method
+	// is *not* used for computing the hash of a leaf that does not exist
+	// (instead, HashEmpty(treeID, index, 0) is used), as the hash value
+	// can be different between:
+	//  - a leaf that was previously set but now has been set to a zero
```

Comment:
can be different between
- A leaf that is unset. 
- A leaf explicitly set, including set to []byte{}.

Reply:
Done.